### DISCUSSION TITLE
Why RenderOptions.BitmapInterpolationMode is not available in xaml?

### DISCUSSION THREAD

**Discussion by J0nathan550** [AvaloniaUI/Avalonia#15387] [Q&A] [answered]
Why `RenderOptions.BitmapInterpolationMode` is not available in xaml why it is hidden but available in the source code?

**Comment by stevemonaco** [comment]
It is available, but IntelliSense won't give any autocomplete.

```xml
<Image Source="..." RenderOptions.BitmapInterpolationMode="None" />
```

**Reply by J0nathan550** [reply]
~It doesn't compile.~
Actually it does compile, but it doesn't work for image brush.

**Reply by yll690** [reply]
Really? I cannot complie and found the BitmapInterpolationMode property is no longer avalible since last year(https://github.com/AvaloniaUI/Avalonia/commit/0442c3b337de3dba8d0f2efa5b2814da0d9050d9#diff-30146ad647035810a03aa11110124ad0fadea7511ded3ed0c037f2e2c3fe26f3L10)
Did you use 0.10? I am using the 11.1.3

**Reply by stevemonaco** [reply]
`RenderOptions.BitmapInterpolationMode` still works in 11.1.3. You won't see it pop up in IntelliSense (or in Rider).

**Reply by yll690** [reply]
Okey, after some attempts, I found it can only set directly to the control instead of setting in styles now.

**Reply by stevemonaco** [reply]
It's a normal property now instead of a `StyledProperty`, so you can't set via style anymore.

**Comment by J0nathan550** [comment]
> It is available, but IntelliSense won't give any autocomplete.
> 
> ```
> <Image Source="..." RenderOptions.BitmapInterpolationMode="None" />
> ```

Actually, it does work. But only for the Image. It doesn't work for ImageBrush.

**Reply by stevemonaco** [reply]
I think it's only valid on `Visual`s. Which `Control`s are and `Brush`es aren't. Otherwise, show more code of what you're trying to do.

**Reply by J0nathan550** [reply]
![image](https://github.com/AvaloniaUI/Avalonia/assets/88881607/2517ec44-5aae-4a5b-a7aa-1f04f159aeb6)
Trying to increase quality of image brush.

**Reply by workgroupengineering** [reply]
there is a [pr](https://github.com/AvaloniaUI/Avalonia/pull/13552) but it was not merged

**Reply by stevemonaco** [reply, marked as answer]
If you set `RenderOptions.BitmapInterpolationMode="HighQuality"` on the `UserControl`, does it affect the brush? There was some talk about allowing overriding the defaults awhile back, but that hasn't been implemented. It's a bit hard to track down.

Alternatively, use layer stacking instead of `ImageBrush`:
```xml
<UserControl ...>
  <Panel>
    <Image Source="/Assets/background.jpg" Stretch="UniformToFill" RenderOptions.BitmapInterpolationMode="HighQuality" />
    <TextBlock Text="Your Stacked Content Here" />
  </Panel>
</UserControl>
```

**Comment by timunie** [comment]
I've a pending PR on that #13552 but I need a 👍 or 👎 from the team

/cc @Gillibald
